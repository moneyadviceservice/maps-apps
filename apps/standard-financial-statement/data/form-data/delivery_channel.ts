export const deliveryChannel = [
  {
    key: 'online',
    en: 'Online',
    cy: 'Ar-lein',
  },
  {
    key: 'telephone',
    en: 'Telephone',
    cy: 'Ff√¥n',
  },
  {
    key: 'face-to-face',
    en: 'Face to Face',
    cy: 'Wyneb yn wyneb',
  },
  // {
  //   key: 'other',
  //   en: 'Other',
  //   cy: 'Arall (Nodwch isod)',
  // },
];
