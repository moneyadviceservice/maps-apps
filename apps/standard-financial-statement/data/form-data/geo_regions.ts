export const geoRegions = [
  {
    key: 'north-west',
    en: 'North West',
    cy: 'Gogledd-orllewin Lloegr',
  },
  {
    key: 'north-east',
    en: 'North East',
    cy: 'Gogledd-ddwyrain Lloegr',
  },
  {
    key: 'yorkshire-and-the-humber',
    en: 'Yorkshire and the Humber',
    cy: "Swydd Efrog a'r Humber",
  },
  {
    key: 'west-midlands',
    en: 'West Midlands',
    cy: 'Gorllewin Canolbarth Lloegr',
  },
  {
    key: 'east-of-england',
    en: 'East of England',
    cy: 'Dwyrain Lloegr',
  },
  {
    key: 'south-west',
    en: 'South West',
    cy: 'De-orllewin Lloegr',
  },
  {
    key: 'london',
    en: 'London',
    cy: 'Llundain',
  },
  {
    key: 'scotland',
    en: 'Scotland',
    cy: 'Yr Alban',
  },
  {
    key: 'south-east',
    en: 'South East',
    cy: 'De-ddwyrain Lloegr',
  },
  {
    key: 'northern-ireland',
    en: 'Northern Ireland',
    cy: 'Gogledd Iwerddon',
  },
  {
    key: 'wales',
    en: 'Wales',
    cy: 'Cymru',
  },
];
