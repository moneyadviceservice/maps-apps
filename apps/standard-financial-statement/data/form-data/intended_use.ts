export const intendedUse = [
  {
    value: 'provide-debt-advice',
    en: 'Provide debt advice',
    cy: 'Darparu cyngor ar ddyledion',
  },
  {
    value: 'collection-of-debt',
    en: 'Collection of debt',
    cy: 'Casglu dyled',
  },
  {
    value: 'insolvency',
    en: 'Insolvency',
    cy: 'Ansolfedd',
  },
  {
    value: 'tech',
    en: 'Tech',
    cy: 'Technoleg',
  },
  {
    value: 'other',
    en: 'Other',
    cy: 'Arall',
  },
];
