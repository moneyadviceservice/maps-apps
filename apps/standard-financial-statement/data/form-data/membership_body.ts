export const membershipBody = [
  {
    key: 'advice-ni',
    en: 'Advice NI',
    cy: 'Cyngor NI',
  },
  {
    key: 'advice-uk',
    en: 'Advice UK',
    cy: 'Cyngor DU',
  },
  {
    key: 'citizens-advice',
    en: 'Citizens Advice',
    cy: 'Cyngor y Bobl',
  },
  {
    key: 'ccua',
    en: 'CCUA',
    cy: 'CCUA',
  },
  {
    key: 'civea',
    en: 'CIVEA',
    cy: 'CIVEA',
  },
  {
    key: 'cma',
    en: 'CMA',
    cy: 'CMA',
  },
  {
    key: 'csa',
    en: 'CSA',
    cy: 'CSA',
  },
  {
    key: 'demsa',
    en: 'DEMSA',
    cy: 'DEMSA',
  },
  {
    key: 'drf',
    en: 'DFR',
    cy: 'DFR',
  },
  {
    key: 'fla',
    en: 'FLA',
    cy: 'FLA',
  },
  {
    key: 'hceoa',
    en: 'HCEOA',
    cy: 'HCEOA',
  },
  {
    key: 'ima',
    en: 'IMA',
    cy: 'IMA',
  },
  {
    key: 'ipa',
    en: 'IPA',
    cy: 'IPA',
  },
  {
    key: 'irrv',
    en: 'IRRV',
    cy: 'IRRV',
  },
  {
    key: 'r3',
    en: 'R3',
    cy: 'R3',
  },
  {
    key: 'uk-finance',
    en: 'UK Finance',
    cy: 'UK Finance',
  },
  {
    key: 'none',
    en: 'none',
    cy: 'none',
  },
  {
    key: 'other',
    en: 'Other (please specify below)',
    cy: 'Arall (Nodwch isod osod)',
  },
];

export const orderedMembershipBody = membershipBody?.sort((a, b) => {
  if (a.key === 'other') return 1;
  if (b.key === 'other') return -1;
  return a.key.localeCompare(b.key);
});
