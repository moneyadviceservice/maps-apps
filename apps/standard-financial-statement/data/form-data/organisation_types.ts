export const organisationTypes = [
  {
    value: 'Free-to-client advice services provider',
    en: 'Free-to-client advice services provider',
    cy: "Darparwr gwasanaethau cynghori am ddim i'r cleient",
  },
  {
    value: 'Central Government department',
    en: 'Central Government department',
    cy: 'Adran Llywodraeth ganolog',
  },
  {
    value: 'Commercial credit provider (including mortgage lender)',
    en: 'Commercial credit provider (including mortgage lender)',
    cy: 'Darparwr credyd masnachol (yn cynnwys darparwr morgais)',
  },
  {
    value: 'Credit union / Community development finance institution',
    en: 'Credit union / Community development finance institution',
    cy: 'Undeb credyd / Sefydliad cyllid datblygu cymunedol',
  },
  {
    value: 'Fuel provider',
    en: 'Fuel provider',
    cy: 'Darparwr tanwydd',
  },
  {
    value: 'Insolvency practitioner',
    en: 'Insolvency practitioner',
    cy: 'Ymarferydd ansolfedd',
  },
  {
    value: 'Debt management company',
    en: 'Debt management company',
    cy: 'Cwmni rheoli dyledion',
  },
  {
    value: 'Local authority',
    en: 'Local authority',
    cy: 'Awdurdod lleol',
  },
  {
    value: 'Registered social landlord',
    en: 'Registered social landlord',
    cy: 'Landlord cymdeithasol cofrestredig',
  },
  {
    value: 'Water supplier',
    en: 'Water supplier',
    cy: 'Cyflenwr d≈µr',
  },
  {
    value: 'Debt collection agency',
    en: 'Debt collection agency',
    cy: 'Asiantaeth casglu dyledion',
  },
  {
    value: 'Software provider',
    en: 'Software provider',
    cy: 'Darparwr meddalwedd',
  },
  {
    value: 'Solicitors',
    en: 'Solicitors',
    cy: 'Cyfreithwyr',
  },
  {
    value: 'other',
    en: 'Other (Please specify below)',
    cy: 'Arall (Nodwch isod)',
  },
];
