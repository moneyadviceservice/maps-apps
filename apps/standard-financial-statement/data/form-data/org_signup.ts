import { z } from 'zod';

import { Question } from '@maps-react/form/types';
import useTranslation from '@maps-react/hooks/useTranslation';

import { deliveryChannel } from './delivery_channel';
import { geoRegions } from './geo_regions';
import { intendedUse } from './intended_use';
import { orderedMembershipBody } from './membership_body';
import { organisationTypes } from './organisation_types';

export interface QuestionOrg extends Question {
  name: string;
  hint?: string;
}

export enum FormType {
  NEW_ORG = 'new',
  NEW_ORG_USER = 'user',
  ACTIVE_ORG = 'active',
  OTP = 'otp',
  SUCCESS = 'success',
}

export const signUpType = (z: ReturnType<typeof useTranslation>['z']) => {
  return [
    { text: z({ en: 'Yes', cy: 'Ydw' }), value: FormType.ACTIVE_ORG },
    { text: z({ en: 'No', cy: 'Nac ydw' }), value: FormType.NEW_ORG },
  ];
};

export const requiredError = {
  en: 'This value is required.',
  cy: `Mae angen y gwerth hwn.`,
};

export const orgAddressQuestions = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg[] => {
  return [
    {
      questionNbr: 1,
      name: 'organisationName',
      group: '',
      title: z({
        en: 'Organisation name',
        cy: `Enw’r sefydliad`,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 2,
      name: 'organisationWebsite',
      group: '',
      title: z({
        en: 'Organisation website address (optional)',
        cy: `Cyfeiriad gwefan y sefydliad (dewisol)`,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
        invalid_string: z({
          en: 'Please enter a valid URL.',
          cy: `Rhowch gyfeiriad gwe dilys.`,
        }),
        invalid_format: z({
          en: 'Please enter a valid URL.',
          cy: `Rhowch gyfeiriad gwe dilys.`,
        }),
      },
    },
    {
      questionNbr: 3,
      name: 'organisationStreet',
      group: '',
      title: z({
        en: 'Organisation street address',
        cy: `Cyfeiriad stryd y sefydliad`,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 4,
      name: 'organisationCity',
      group: '',
      title: z({
        en: 'Organisation town/city',
        cy: `Tref/dinas y sefydliad`,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 5,
      name: 'organisationPostcode',
      group: '',
      title: z({
        en: 'Organisation postcode',
        cy: `Cod post y sefydliad`,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
  ];
};

export const orgType = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'organisationType',
    group: '',
    title: z({
      en: 'What type of organisation are you?',
      cy: `Pa fath o sefydliad ydych chi?`,
    }),
    type: 'select',
    answers: organisationTypes.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.value,
    })),
    errors: {
      invalid_type: z({
        en: 'This value is required.',
        cy: `Rhowch enw sefydliad dilys.`,
      }),
      too_small: z(requiredError),
    },
  };
};

export const orgLive = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'sfslive',
    group: '',
    title: z({
      en: 'Are you live with the SFS?',
      cy: `A ydych yn gwneud defnydd byw o'r SFS?`,
    }),
    type: 'radio',
    answers: [
      { text: z({ en: 'Yes', cy: 'Ydw' }), value: 'true' },
      { text: z({ en: 'No', cy: 'Nac ydw' }), value: 'false' },
    ],
    errors: {
      too_small: z(requiredError),
    },
  };
};

const orderedGeoRegions = geoRegions?.sort((a, b) =>
  a.key.localeCompare(b.key),
);

export const orgGeoRegions = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'geoRegions',
    group: '',
    title: z({
      en: 'What geographical regions does your services cover?',
      cy: `Pa ranbarthau daearyddol sy'n cael eu cynnwys gan eich gwasanaethau?`,
    }),
    description: z({
      en: 'Select all that apply',
      cy: `Ticiwch bob un sy'n berthnasol`,
    }),
    type: 'checkbox',
    answers: orderedGeoRegions.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.key,
    })),
    errors: {
      invalid_type: z({
        en: 'Please select type',
        cy: `Rhowch enw sefydliad dilys.`,
      }),
      too_small: z(requiredError),
    },
  };
};

export const orgUse = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'organisationUse',
    group: '',
    title: z({
      en: 'What is your intended use of SFS?',
      cy: `Beth yw’ch defnydd arfaethedig o SFS?`,
    }),
    type: 'select',
    answers: intendedUse.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.value,
    })),
    errors: {
      too_small: z(requiredError),
    },
  };
};

export const debtAdvice = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'debtAdvice',
    group: '',
    title: z({
      en: 'How do you deliver advice?',
      cy: `Sut ydych chi’n darparu cyngor?`,
    }),
    type: 'checkbox',
    answers: deliveryChannel.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.key,
    })),
    errors: {
      too_small: z(requiredError),
      invalid_type: z({
        en: 'This value is required.',
        cy: `Rhowch enw sefydliad dilys.`,
      }),
    },
  };
};

export const debtAdviceOther = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'debtAdviceOther',
    group: '',
    title: '',
    type: 'single',
    answers: [],
    errors: {
      too_small: z(requiredError),
      invalid_type: z({
        en: 'This value is required.',
        cy: `Rhowch enw sefydliad dilys.`,
      }),
    },
  };
};

export const orgLaunchDate = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'sfsLaunchDate',
    group: '',
    title: z({
      en: 'SFS Launch date (or estimated launch date)',
      cy: `Dyddiad lawnsio SFS (neu ddyddiad lawnsio arfaethedig)`,
    }),
    type: 'date',
    answers: intendedUse.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.value,
    })),
    errors: {
      invalid_type: z({
        en: 'Please enter a valid date.',
        cy: `Rhowch dyddiad dilys.`,
      }),
      too_small: z(requiredError),
    },
  };
};

export const orgSoftware = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'caseManagementSoftware',
    group: '',
    title: z({
      en: 'Name of case management software used (if applicable)',
      cy: `Enw’r meddalwedd rheoli achosion a ddefnyddir (os yn berthnasol)`,
    }),
    type: 'text',
    answers: intendedUse.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.value,
    })),
    errors: {
      invalid_type: z({
        en: 'Please select type',
        cy: `Rhowch enw sefydliad dilys.`,
      }),
      too_small: z(requiredError),
    },
  };
};

export const orgFcaReg = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'fcaReg',
    group: '',
    title: z({
      en: 'Is your organisation registered with the Financial Conduct Authority?',
      cy: `A yw eich sefydliad wedi'i gofrestru gyda'r Awdurdod Ymddygiad Ariannol?`,
    }),
    type: 'radio',
    answers: [
      { text: z({ en: 'Yes', cy: 'Ydw' }), value: 'fca-yes' },
      { text: z({ en: 'No', cy: 'Nac ydw ' }), value: 'fca-no' },
    ],
    errors: {
      too_small: z(requiredError),
    },
  };
};

export const orgFcaRegNumber = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'fcaRegNumber',
    group: '',
    title: z({
      en: 'Please provide your FCA License number',
      cy: `Rhowch eich rhif Trwydded FCA`,
    }),
    type: 'single',
    answers: [],
    errors: {
      too_small: z(requiredError),
    },
  };
};

export const membershipBodyOpt = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'memberships',
    group: '',
    title: z({
      en: 'If you belong to a trade or membership body, please select from the list below:',
      cy: `Os ydych chi'n perthyn i gorff masnachu neu aelodaeth, dewiswch o'r rhestr isod:`,
    }),
    type: 'single',
    answers: orderedMembershipBody.map((type) => ({
      text: z({
        en: type.en,
        cy: type.cy,
      }),
      value: type.key,
    })),
    errors: {
      too_small: z(requiredError),
    },
  };
};

export const allQuestions = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg[] => {
  return [
    ...orgAddressQuestions(z),
    orgLaunchDate(z),
    orgSoftware(z),
    orgFcaReg(z),
    orgFcaRegNumber(z),
    membershipBodyOpt(z),
    orgType(z),
    orgLive(z),
    orgGeoRegions(z),
    orgUse(z),
    debtAdvice(z),
    debtAdviceOther(z),
    ...orderedMembershipBody.map((member) => ({
      questionNbr: 1,
      name: member.key,
      group: '',
      title: z({
        en: member.en,
        cy: member.cy,
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    })),
  ];
};
