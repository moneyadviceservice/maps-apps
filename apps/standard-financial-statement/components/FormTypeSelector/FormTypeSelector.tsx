import { ChangeEvent } from 'react';

import { radioStyles } from 'components/RichTextWrapper';
import { FormType, signUpType } from 'data/form-data/org_signup';
import { twMerge } from 'tailwind-merge';

import { RadioButton } from '@maps-react/form/components/RadioButton';
import useTranslation from '@maps-react/hooks/useTranslation';

export const FormTypeSelector = ({
  flow,
  onChange,
}: {
  flow: FormType;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}) => {
  const { z } = useTranslation();
  const input = signUpType(z);

  return (
    <fieldset className="relative">
      <legend className="mb-4 text-2xl text-gray-900">
        {z({
          en: 'Is your organisation a member of SFS?',
          cy: 'Ydyâ€™ch sefydliad yn aelod o SFS?',
        })}
      </legend>
      {input.map((answer) => (
        <div key={`radio-${answer.text}`} className="mb-4">
          <RadioButton
            name="sfs-status"
            id={`${answer.value}`}
            value={answer.value}
            defaultChecked={flow === answer.value}
            classNameLabel={twMerge(radioStyles)}
            onChange={onChange}
            hasError={false}
          >
            {answer.text}
          </RadioButton>
        </div>
      ))}
    </fieldset>
  );
};
