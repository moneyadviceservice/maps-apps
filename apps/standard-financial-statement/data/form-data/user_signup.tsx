import { Link } from '@maps-react/common/index';
import useTranslation from '@maps-react/hooks/useTranslation';

import { FormType, QuestionOrg, requiredError } from './org_signup';

export const orgLicenceNumber = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'orgLicenceNumber',
    group: '',
    title: z({
      en: 'Membership number:',
      cy: 'Rhif aelodaeth:',
    }),
    type: 'text',
    answers: [],
    errors: {
      too_small: z(requiredError),
      not_found: z({
        en: 'Organisation not found',
        cy: 'Sefydliad heb ei ddarganfod',
      }),
    },
  };
};

export const codeOfConduct = (
  z: ReturnType<typeof useTranslation>['z'],
): QuestionOrg => {
  return {
    questionNbr: 1,
    name: 'codeOfConduct',
    group: '',
    title: z({
      en: 'Code of Conduct agreement',
      cy: 'Cytundeb Cod Ymddygiad',
    }),
    definition: z({
      en: (
        <>
          I agree to the{' '}
          <Link
            href="/en/apply-to-use-the-sfs/sfs-code-of-conduct"
            className="font-bold text-magenta-800"
            target="_blank"
            withIcon={false}
          >
            Code of Conduct
          </Link>
        </>
      ),
      cy: (
        <>
          Rwy’n cytuno i’r{' '}
          <Link
            href="/cy/apply-to-use-the-sfs/sfs-code-of-conduct"
            className="font-bold text-magenta-800"
            target="_blank"
            withIcon={false}
          >
            Cod Ymddygiad
          </Link>
        </>
      ),
    }),
    description: z({
      en: 'Please tick this box to confirm you agree to the SFS Code of Conduct, which provides a guide to best practice use of the SFS.',
      cy: 'Ticiwch y blwch hwn i gadarnhau eich bod yn cytuno i God Ymddygiad SFS, sy’n darparu canllaw i arfer gorau o ddefnyddio’r SFS.',
    }),
    type: 'checkbox',
    answers: [],
    errors: {
      custom: z(requiredError),
    },
  };
};

export const userForm = (
  z: ReturnType<typeof useTranslation>['z'],
  formType: FormType,
): QuestionOrg[] => {
  return [
    formType === FormType.ACTIVE_ORG ? orgLicenceNumber(z) : null,
    {
      questionNbr: 1,
      name: 'firstName',
      group: '',
      title: z({
        en: 'First name',
        cy: 'Enw cyntaf',
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 2,
      name: 'lastName',
      group: '',
      title: z({
        en: 'Surname',
        cy: 'Cyfenw',
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 3,
      name: 'emailAddress',
      group: '',
      title: z({
        en: 'Email address',
        cy: 'Cyfeiriad e-bost',
      }),
      hint: z({
        en: 'Enter your organisational email',
        cy: 'Rhowch eich e-bost sefydliadol',
      }),
      type: 'email',
      answers: [],
      errors: {
        too_small: z(requiredError),
        invalid_string: z({
          en: 'Invalid email address',
          cy: 'Cyfeiriad e-bost annilys',
        }),
        user_already_exists: z({
          en: 'An account with this email address already exists. Sign in or use a different email address.',
          cy: 'Mae cyfrif gyda’r cyfeiriad e-bost hwn yn bodoli eisoes. Mewngofnodwch neu defnyddiwch gyfeiriad e-bost gwahanol.',
        }),
        not_allowed: z({
          en: 'The email address you entered is not from your organisation. Please use your organisational email.',
          cy: 'Nid yw’r cyfeiriad e-bost rydych chi wedi’i roi yn dod o’ch sefydliad. Rhowch eich e-bost sefydliad.',
        }),
      },
    },
    {
      questionNbr: 4,
      name: 'tel',
      group: '',
      title: z({
        en: 'Telephone Number',
        cy: 'Rhif ffôn',
      }),
      type: 'number',
      answers: [],
      errors: {
        too_small: z(requiredError),
        invalid_type: z({
          en: 'Invalid number',
          cy: 'invalid number',
        }),
      },
    },
    {
      questionNbr: 5,
      name: 'jobTitle',
      group: '',
      title: z({
        en: 'Job Title',
        cy: 'Teitl y swydd',
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
      },
    },
    {
      questionNbr: 6,
      name: 'password',
      hint: z({
        en: 'Enter a password that is 8 to 20 characters long and includes at least one uppercase letter, one lowercase letter, and one special character (for example, !@#$%&)',
        cy: 'Rhowch gyfrinair sydd rhwng 8 ac 20 nod a’n cynnwys o leiaf un briflythyren, un llythyren fach, ac un nod arbennig (er enghraifft, !@#$%&)',
      }),
      group: '',
      title: z({
        en: 'Password',
        cy: 'Cyfrinair',
      }),
      type: 'password',
      answers: [],
      errors: passwordErrors(z),
    },
    {
      questionNbr: 7,
      name: 'confirmPassword',
      group: '',
      title: z({
        en: 'Confirm Password',
        cy: 'Cadarnhau cyfrinair',
      }),
      type: 'password',
      answers: [],
      errors: passwordErrors(z),
    },
    codeOfConduct(z),
  ].filter(Boolean) as QuestionOrg[];
};

const passwordErrors = (z: ReturnType<typeof useTranslation>['z']) => ({
  too_small: z(requiredError),
  invalid_type: z({
    en: 'Please ensure your password and password confirmation are identical.',
    cy: 'Sicrhewch fod eich cyfrinair a chadarnhad cyfrinair yn unfath.',
  }),
  custom: z({
    en: 'Enter a password that is 8 to 20 characters long and includes at least one uppercase letter, one lowercase letter, and one special character (for example, !@#$%&)',
    cy: 'Rhowch gyfrinair sydd rhwng 8 ac 20 nod a’n cynnwys o leiaf un briflythyren, un llythyren fach, ac un nod arbennig (er enghraifft, !@#$%&)',
  }),
});

export const userFormOTP = (
  z: ReturnType<typeof useTranslation>['z'],
  emailAddress?: string,
): QuestionOrg[] | null[] => {
  return [
    {
      questionNbr: 1,
      name: 'otp',
      group: '',
      title: z({
        en: 'One-time passcode',
        cy: 'Cod mynediad untro',
      }),
      description: z({
        en: 'Please enter the One-time passcode which has been sent to:',
        cy: 'Rhowch y Cod Mynediad Untro sydd wedi’i anfon at:',
      }),
      type: 'text',
      answers: [],
      errors: {
        too_small: z(requiredError),
        invalid_grant: z({
          en: 'Incorrect one-time passcode. Please check the code and try again.',
          cy: 'Rhowch y Cod Mynediad Untro sydd wedi’i anfon at:',
        }),
        expired_token: z({
          en: `This one-time passcode has expired. We have sent a new code to ${emailAddress}. Please enter the new code below.`,
          cy: `Mae’r Cod Mynediad Untro hwn wedi dod i ben. Rydym wedi anfon cod newydd at ${emailAddress}. Rhowch y cod newydd isod.`,
        }),
      },
    },
  ];
};
