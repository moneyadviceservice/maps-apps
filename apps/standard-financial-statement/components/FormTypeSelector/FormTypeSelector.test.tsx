import { FormType } from 'data/form-data/org_signup';
import { render } from '@testing-library/react';

import { FormTypeSelector } from './FormTypeSelector';

import '@testing-library/jest-dom';

jest.mock('@maps-react/hooks/useTranslation', () => () => ({
  z: ({ en }: { en: string; cy: string }) => en,
}));

describe('FormTypeSelector', () => {
  it('renders correctly as new org', () => {
    const { container } = render(
      <FormTypeSelector flow={FormType.NEW_ORG} onChange={jest.fn()} />,
    );
    expect(container).toMatchSnapshot();
  });
});
