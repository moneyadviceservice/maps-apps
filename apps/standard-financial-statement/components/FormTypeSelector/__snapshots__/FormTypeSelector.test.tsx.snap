// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTypeSelector renders correctly as new org 1`] = `
<div>
  <fieldset
    class="relative"
  >
    <legend
      class="mb-4 text-2xl text-gray-900"
    >
      Is your organisation a member of SFS?
    </legend>
    <div
      class="mb-4"
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="active"
          name="sfs-status"
          type="radio"
          value="active"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-disabled:active:after:border-gray-200 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100 peer-checked:after:border-green-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-green-500 peer-focus:active:after:border-green-500 peer-checked:hover:after:border-green-500 peer-focus:hover:after:border-green-500""
          for="active"
        >
          Yes
        </label>
      </div>
    </div>
    <div
      class="mb-4"
    >
      <div
        class="relative flex flex-wrap items-center text-base"
        data-testid="radio-button"
      >
        <input
          checked=""
          class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
          id="new"
          name="sfs-status"
          type="radio"
          value="new"
        />
        <label
          class="align-self-center max-w-[calc(100%-40px)] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-magenta-500 after:rounded-full after:opacity-0 peer-disabled:active:after:border-gray-200 peer-focus:before:border-4 peer-focus:before:border-blue-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_3px] peer-focus:before:shadow-yellow-400 peer-checked:after:opacity-100 peer-checked:after:border-green-500 active:after:opacity-100 active:after:border-white peer-checked:active:after:border-green-500 peer-focus:active:after:border-green-500 peer-checked:hover:after:border-green-500 peer-focus:hover:after:border-green-500""
          for="new"
        >
          No
        </label>
      </div>
    </div>
  </fieldset>
</div>
`;
